src/app/admin/adminApp.js

```diff
@@ -11,44 +11,39 @@ import {
 import { dataProvider } from "ra-data-simple-prisma";
 
 const authProvider = ({ signIn, signOut }) => ({
-  login: async (credentials) => {
-    console.log("login", credentials);
-    return signIn(credentials);
-  },
-  logout: () => {
-    console.log("logout");
-    return signOut();
-  },
-  checkError: ({ status }) => Promise.resolve(),
-  // checkError: ({ status }) => {
-  // if (status === 401 || status === 403) {
-  //   localStorage.removeItem("username");
-  //   return Promise.reject();
-  // }
-  // return Promise.resolve();
-  // },
-  // called when the user navigates to a new location, to check for authentication
-  checkAuth: () => Promise.resolve(),
-  // checkAuth: () => {
-  // return Promise.resolve();
-  // console.log("checkAuth");
-  // return localStorage.getItem("username")
-  //   ? Promise.resolve()
-  //   : Promise.reject();
-  // },
-  getPermissions: () => Promise.resolve(),
+    login: async (credentials) => {
+      await signIn(credentials);
+      return Promise.resolve();
+    },
+    logout: async () => {
+      await signOut();
+      return Promise.resolve();
+    },
+    checkAuth: async () => {
+      const session = await getSession();
+      if (session === null) {
+        return Promise.reject();
+      }
+      return Promise.resolve();
+    },
+    checkError:  (error) => Promise.resolve(),
```

Comment:
is it required to define checkError function?
if not,  i do not see any reason to define it and return Promise.resolve() especially if you do not put async word before function definition

Reply:
checkError and getPermissions functions are used by react admin, and must have a definition